src/services/transaction_reversal.rs

```diff
@@ -15,17 +20,200 @@ pub struct TransactionReversalPayload<'mpesa> {
     #[serde(rename(serialize = "ReceiverParty"))]
     receiver_party: &'mpesa str,
     #[serde(rename(serialize = "ReceiverIdentifierType"))]
-    receiver_identifier_type: &'mpesa str,
+    receiver_identifier_type: Option<&'mpesa str>,
     #[serde(rename(serialize = "ResultURL"))]
-    result_url: &'mpesa str,
+    result_url: Option<&'mpesa str>,
     #[serde(rename(serialize = "QueueTimeOutURL"))]
-    queue_timeout_url: &'mpesa str,
+    timeout_url: Option<&'mpesa str>,
     #[serde(rename(serialize = "Remarks"))]
-    remarks: &'mpesa str,
+    remarks: Option<&'mpesa str>,
     #[serde(rename(serialize = "Occasion"))]
-    ocassion: &'mpesa str,
+    occasion: Option<&'mpesa str>,
+    #[serde(rename(serialize = "Amount"))]
+    amount: f64,
 }
 
-pub struct TransactionReversalBuilder;
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TransactionReversalResponse {
+    #[serde(rename(deserialize = "ConversationID"))]
+    conversation_id: String,
+    #[serde(rename(deserialize = "OriginatorConversationID"))]
+    originator_conversation_id: String,
+    #[serde(rename(deserialize = "ResponseDescription"))]
+    response_description: String,
+}
+
+/// Creates new `TransactionReversalBuilder`
```

Comment:
I think this comment should be on the builder's `new` method

Reply:
True